src/common/role-components.js

```diff
@@ -16,6 +16,6 @@ export const RoleComponentsBehavior = {
   },
 
   _hasComponent(comp) {
-    return this.components.length === 0 || this.components.includes(comp);
+    return this.components.includes("all") || this.components.includes(comp);
```

Comment:
I feel like this approach is more "mistake proof". In case someone forgets to configure the permissions for a role it should not get access to everything, the permissions for accessing everything should be given knowingly.
I have discussed this this approach with @bogdan-sava and we agreed that this is a better handling of the "admin" role.

Reply:
@marius-patrascu maybe I was not clear.
the "all" keyword is to be used only in backend app configuration.
Is meant to return an empty array if present.

If current user has more roles and one of them has "all", or maybe "all-components" means that API response is an empty array, and UI knows what that means.

So, the backend add all roles arrays except one of them contains "all-components" when returns empty array.

Reply:
as @sarbull suggested "*" is even better :)